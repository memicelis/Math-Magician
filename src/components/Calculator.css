body {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.calculator-container {
  display: flex;
  justify-content: space-between;
}

.calculator-container h2 {
  font-size: 2.5rem;
  padding: 20px;
}

.calculator {
  padding: 50px;
  width: 30rem;
}

.display {
  display: flex;
}

.display input {
  width: 100%;
  height: 60px;
  border: 1px solid #8f8f8f;
  background: #b5b5bd;
  color: #fff;
  text-align: right;
  font-size: 36px;
}

.buttons {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(5, 60px);
}

.calculator-button {
  border: 1px solid #8f8f8f;
  font-size: 24px;
}

.calculator-button:nth-child(4n),
.calculator-button:last-child {
  background-color: orange;
}

.calculator-button--0 {
  grid-column: span 2;
}
