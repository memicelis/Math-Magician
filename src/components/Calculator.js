import React from 'react';
import './Calculator.css';
import Clear from './Clear';

const Calculator = () => (
  <div className="calculator">
    <div className="display">
      <input className="input-display" type="text" />
    </div>
    <div className="buttons">
      <Clear />
      <button className="calculator-button" type="button">
        +/-
      </button>
      <button className="calculator-button" type="button">
        %
      </button>
      <button className="calculator-button" type="button">
        รท
      </button>
      <button className="calculator-button" type="button">
        7
      </button>
      <button className="calculator-button" type="button">
        8
      </button>
      <button className="calculator-button" type="button">
        9
      </button>
      <button className="calculator-button" type="button">
        x
      </button>
      <button className="calculator-button" type="button">
        4
      </button>
      <button className="calculator-button" type="button">
        5
      </button>
      <button className="calculator-button" type="button">
        6
      </button>
      <button className="calculator-button" type="button">
        -
      </button>
      <button className="calculator-button" type="button">
        1
      </button>
      <button className="calculator-button" type="button">
        2
      </button>
      <button className="calculator-button" type="button">
        3
      </button>
      <button className="calculator-button" type="button">
        +
      </button>
      <button className="calculator-button calculator-button--0" type="button">
        0
      </button>
      <button className="calculator-button" type="button">
        .
      </button>
      <button className="calculator-button" type="button">
        =
      </button>
    </div>
  </div>
);

export default Calculator;
