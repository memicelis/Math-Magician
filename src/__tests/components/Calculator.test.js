import React from 'react';
import Calculator from '../../components/Calculator';
import renderer from 'react-test-renderer';

describe('testing Calculator component rendering', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
