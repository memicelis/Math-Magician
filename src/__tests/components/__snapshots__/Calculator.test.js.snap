// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Calculator component rendering renders correctly 1`] = `
<div
  className="calculator-container"
>
  <h2>
    Let\`s do Some Math
  </h2>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      <input
        className="input-display"
        type="text"
        value=""
      />
    </div>
    <div
      className="buttons"
    >
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="calculator-button calculator-button--0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calculator-button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
